src/os/layer/preset/presetmenubutton.js

```diff
@@ -0,0 +1,220 @@
+          icons: ['<i class="fa fa-fw fa-eye-slash"></i>'],
+          handler: this.togglePublished.bind(this),
+          metricKey: Metrics.TOGGLE_PUBLISHED,
+          sort: 11
+        }, {
+          label: 'Set as Default',
+          eventType: EventType.TOGGLE_DEFAULT_TRUE,
+          tooltip: 'Make Preset the default style when layer added',
+          icons: ['<i class="fa fa-fw fa-star"></i>'],
+          handler: this.toggleDefault.bind(this),
+          metricKey: Metrics.TOGGLE_DEFAULT,
+          sort: 20
+        }, {
+          label: 'Revoke Default',
+          eventType: EventType.TOGGLE_DEFAULT_FALSE,
+          tooltip: 'Undo making Preset the default style when layer added',
+          icons: ['<i class="fa fa-fw fa-star-o"></i>'],
+          handler: this.toggleDefault.bind(this),
+          metricKey: Metrics.TOGGLE_DEFAULT,
+          sort: 21
+        }, {
+          label: 'Delete',
+          eventType: EventType.REMOVE,
+          tooltip: 'Permanently delete Preset',
+          icons: ['<i class="fa fa-fw fa-close"></i>'],
+          handler: this.remove.bind(this),
+          metricKey: Metrics.REMOVE,
+          sort: 30
+        }]
+      }]
+    }));
+    this.flag = MENU_FLAG;
+    this.metricKey = Metrics.OPEN;
+  }
+
+  /**
+   * Angular $onDestroy lifecycle hook.
+   */
+  $onDestroy() {
+    this.scope = null;
+  }
+
+  /**
+   *
+   */
+  remove() {
+    const preset = this.scope['parentCtrl']['preset'];
+
+    console.log(`preset.remove(${preset.label})`);
+  }
+
+  /**
+   *
+   */
+  save() {
+    const preset = this.scope['parentCtrl']['preset'];
+
+    console.log(`preset.save(${preset.label})`, preset);
+  }
+
+  /**
+   *
+   */
+  toggleDefault() {
+    const preset = this.scope['parentCtrl']['preset'];
+
+    console.log(`preset.toggleDefault(${preset.label}), was: ${preset.default}`);
+  }
+
+  /**
+   *
+   */
+  togglePublished() {
+    const preset = this.scope['parentCtrl']['preset'];
+
+    console.log(`preset.togglePublished(${preset.label}), was: ${preset.published}`);
+  }
+
+  /**
+   *
+   */
+  visibility() {
+    const preset = this.scope['parentCtrl']['preset'];
+
+    if (preset) {
+      const publishTrue = this.menu.getRoot().find(EventType.TOGGLE_PUBLISHED_TRUE);
+      const publishFalse = this.menu.getRoot().find(EventType.TOGGLE_PUBLISHED_FALSE);
+      const defaultTrue = this.menu.getRoot().find(EventType.TOGGLE_DEFAULT_TRUE);
+      const defaultFalse = this.menu.getRoot().find(EventType.TOGGLE_DEFAULT_FALSE);
+
+      if (publishTrue) publishTrue.visible = !preset.published;
+      if (publishFalse) publishFalse.visible = preset.published;
+      if (defaultTrue) defaultTrue.visible = !preset.default;
+      if (defaultFalse) defaultFalse.visible = preset.default;
+    }
+  }
+}
+
+/**
+ * The preset directive.
+ * @return {angular.Directive}
+ */
+const directive = () => ({
+  restrict: 'E',
+  replace: true,
+  scope: {
+    'parentCtrl': '='
```

Comment:
Passing the whole `parentCtrl` instance down to here creates an uncomfortably close coupling. I would rather see just the preset getting passed in and then potentially using Angular eventing for other communication between the UIs.

Reply:
Ah yup -- I had just the preset but swapped it around while I was testing things out. I'll put that back.

Reply:
Ty ty, @jsalankey -- I updated it to use the $emit from the child and $on in the parent.